toolchain/check/context.cpp

```diff
@@ -112,6 +114,41 @@ auto Context::NoteIncompleteClass(SemIR::ClassId class_id,
   }
 }
 
+auto Context::AddPackageImports(Parse::Node import_node,
+                                IdentifierId package_id,
+                                llvm::ArrayRef<const SemIR::File*> sem_irs,
+                                bool has_failure) -> void {
+  CARBON_CHECK(has_failure || !sem_irs.empty())
+      << "There should be either a failed import or at least one IR.";
+
+  auto name_id = SemIR::NameId::ForIdentifier(package_id);
+
+  SemIR::CrossReferenceIRId first_id(cross_reference_irs().size());
+  for (const auto* sem_ir : sem_irs) {
+    cross_reference_irs().Add(sem_ir);
+  }
+  if (has_failure) {
+    cross_reference_irs().Add(nullptr);
+  }
+  SemIR::CrossReferenceIRId last_id(cross_reference_irs().size() - 1);
+
+  auto inst_id = AddInst(SemIR::Import{.parse_node = import_node,
+                                       .first_cross_reference_ir_id = first_id,
+                                       .last_cross_reference_ir_id = last_id});
+  if (name_id.is_valid()) {
+    // Add the import to lookup. Should always succeed because imports will be
+    // uniquely named.
+    AddNameToLookup(import_node, name_id, inst_id);
+    // This binding is added primarily to show the name in formatted output.
+    AddInst(SemIR::BindName{.parse_node = import_node,
+                            .type_id = SemIR::TypeId::Error,
+                            .name_id = name_id,
+                            .value_id = inst_id});
```

Comment:
Is this the representation you want in the longer term, or should we add a TODO here to revisit this?